File: ios/FileExportModule.swift
import Foundation
import React

@objc(FileExportModule)
class FileExportModule: NSObject {

  @objc
  func exportLog(_ logData: String, resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) -> Void {
    do {
      // Get the documents directory path
      let documentsPath = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)[0]

      // Create filename with timestamp
      let dateFormatter = DateFormatter()
      dateFormatter.dateFormat = "yyyy-MM-dd_HH-mm-ss"
      let timestamp = dateFormatter.string(from: Date())
      let filename = "agent-activity-log_\(timestamp).txt"

      // Full file path
      let filePath = documentsPath.appendingPathComponent(filename)

      // Write data to file using native file APIs
      try logData.write(to: filePath, atomically: true, encoding: .utf8)

      // Resolve with the file path
      resolver(filePath.path)
    } catch {
      rejecter("FILE_EXPORT_ERROR", "Failed to export log: \(error.localizedDescription)", error)
    }
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}

File: android/app/src/main/java/com/agentcontrolapp/FileExportModule.kt
package com.agentcontrolapp

import android.os.Environment
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import java.io.File
import java.io.FileOutputStream
import java.text.SimpleDateFormat
import java.util.*

class FileExportModule(reactContext: ReactApplicationContext) :
    ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String {
        return "FileExportModule"
    }

    @ReactMethod
    fun exportLog(logData: String, promise: Promise) {
        try {
            // Get the documents directory (external storage)
            val documentsDir = reactApplicationContext.getExternalFilesDir(Environment.DIRECTORY_DOCUMENTS)

            if (documentsDir == null) {
                promise.reject("FILE_EXPORT_ERROR", "Could not access documents directory")
                return
            }

            // Create filename with timestamp
            val dateFormat = SimpleDateFormat("yyyy-MM-dd_HH-mm-ss", Locale.getDefault())
            val timestamp = dateFormat.format(Date())
            val filename = "agent-activity-log_$timestamp.txt"

            // Create file
            val file = File(documentsDir, filename)

            // Write data to file using native file APIs
            FileOutputStream(file).use { outputStream ->
                outputStream.write(logData.toByteArray())
            }

            // Resolve with the file path
            promise.resolve(file.absolutePath)
        } catch (e: Exception) {
            promise.reject("FILE_EXPORT_ERROR", "Failed to export log: ${e.localizedMessage}", e)
        }
    }
}
